compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -279,6 +279,8 @@
     private Stack<TopLevelNode> additionalTopLevelNodes = new Stack<>();
     /* To keep track of additional statements produced from multi-BLangNode resultant transformations */
     private Stack<BLangStatement> additionalStatements = new Stack<>();
+    /* To keep track if we are inside a block statment for the use of type definition creation */
+    private boolean isInLocalContext = false;
```

Comment:
Checking `instanceOf` is far better than this IMO? Having these type of state variables will make the Transformer fragile WDYT @dulvinw @manuranga ?

Reply:
Why we need to track this anyway? while we are traversing through the each statements of the statement list we know we are inside a local context. Can have a private method to create anon record.

Reply:
The other option is to add a big switch case for all the statements. Thought this is better performance wise. And we are creating type descriptors in both local and module level. hence need to differentiate.

Reply:
Can't we use [this](https://github.com/ballerina-platform/ballerina-lang/blob/1fe3d3d4d5f58a805df1ff0b33d1f2d9392499ed/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java#L1880) check to differentiate inner type descriptors?
```java
private List<BLangStatement> generateBLangStatements(NodeList<StatementNode> statementNodes) {
    // ...
   for (StatementNode statement : statementNodes) {
      // if type desc
       createTypeDesc(node, false)
       //
   }
   // ...
}

public transform(TypeDescriptorFoo) {
    createTypeDesc(node, true)
}

private createTypeDesc(node, boolean isInner) {
    if(isInner) {
      // create anon
    } else {
     // create top-level
    }
}
```